Rules/CTF/Scripts/CTF_GiveSpawnItems.as

```diff
@@ -105,90 +57,89 @@ void onPlayerDie(CRules@ this, CPlayer@ victim, CPlayer@ attacker, u8 customData
 {
 	if (victim !is null)
 	{
-		RulesCore@ core;
-		this.get("core", @core);
-		if (core !is null)
-		{
-			CTFPlayerInfo@ info = cast < CTFPlayerInfo@ > (core.getInfoFromPlayer(victim));
-			if (info !is null)
-			{
-				info.items_collected &= ~ItemFlag::Archer;
-			}
-		}
+		SetCTFTimerArcher(this, victim, 0);
 	}
 }
 
-bool canGetSpawnmats(CRules@ this, CPlayer@ p, RulesCore@ core)
+string getCTFTimerPropertyNameBuilder(CPlayer@ p)
 {
-	s32 next_items = getCTFTimer(this, p);
-	s32 gametime = getGameTime();
-
-	CTFPlayerInfo@ info = cast < CTFPlayerInfo@ > (core.getInfoFromPlayer(p));
-
-	if (gametime > next_items)		// timer expired
-	{
-		info.items_collected = 0; //reset available class items
-		return true;
-	}
-	else //trying to get new class items, give a guy a break
-	{
-		u32 items = info.items_collected;
-		u32 flag = 0;
-
-		CBlob@ b = p.getBlob();
-		string name = b.getName();
-		if (name == "builder")
-			flag = ItemFlag::Builder;
-		else if (name == "knight")
-			flag = ItemFlag::Knight;
-		else if (name == "archer")
-			flag = ItemFlag::Archer;
-
-		if (info.items_collected & flag == 0)
-		{
-			return true;
-		}
-	}
+	return SPAWN_ITEMS_TIMER_BUILDER + p.getUsername();
+}
 
-	return false;
+s32 getCTFTimerBuilder(CRules@ this, CPlayer@ p)
+{
+	string property = getCTFTimerPropertyNameBuilder(p);
+	if (this.exists(property))
+		return this.get_s32(property);
+	else
+		return 0;
+}
 
+void SetCTFTimerBuilder(CRules@ this, CPlayer@ p, s32 time)
+{
+	string property = getCTFTimerPropertyNameBuilder(p);
+	this.set_s32(property, time);
+	this.SyncToPlayer(property, p);
 }
 
-string getCTFTimerPropertyName(CPlayer@ p)
+string getCTFTimerPropertyNameArcher(CPlayer@ p)
```

Comment:
imo get rid of these funcs

do getCTFTimerPropertyName, which takes a player and a blob. you can just use the blob name instead of hard coding different string names and funcs

Reply:
The point is to allow for displaying resupply text of a class regardless of what class the player is though (in this case specifically, display builder resupply text even if you're archer or knight. Archer resupply isn't displayed for other classes, but it's done the same way for consistency)

With that, should it still be modified? I could combine the functions into one and just add a string parameter instead of adding a blob parameter (string "archer" or string "builder")?